src/imitation/testing/types.py

```diff
@@ -0,0 +1,31 @@
+"""Helper methods for tests involving imitation.data.types."""
+
+import dataclasses
+from typing import Sequence
+
+import numpy as np
+
+from imitation.data import types
+
+
+def assert_traj_equal(traj_a: types.Trajectory, traj_b: types.Trajectory) -> None:
+    """Assert trajectory `a` and `b` are equal."""
+    dict_a, dict_b = dataclasses.asdict(traj_a), dataclasses.asdict(traj_b)
+    assert dict_a.keys() == dict_b.keys()
+    for k, a_v in dict_a.items():
+        b_v = dict_b[k]
+        if k == "infos":
+            # Treat None equivalent to sequence of empty dicts
+            a_v = [{}] * len(traj_a) if a_v is None else a_v
+            b_v = [{}] * len(traj_b) if b_v is None else b_v
+        assert np.array_equal(a_v, b_v)
+
+
+def assert_traj_sequences_equal(
```

Comment:
Maybe we would not need this if `assert_traj_equal` would be implemented as `__eq__(self, other)` in the trajectory class?

Reply:
Good idea, I've implemented `__eq__` now (plus added a test for it).